h2o-core/src/main/java/water/init/NetworkInit.java

```diff
@@ -38,16 +55,19 @@ public static InetAddress findInetAddressForSelf() throws Error {
     return null;
   }
 
-  // Parse arguments and set cloud name in any case. Strip out "-name NAME"
-  // and "-flatfile <filename>". Ignore the rest. Set multi-cast port as a hash
-  // function of the name. Parse node ip addresses from the filename.
+  /**
+   *  Parse arguments and set cloud name in any case. Strip out "-name NAME"
+   *  and "-flatfile <filename>". Ignore the rest. Set multi-cast port as a hash
+   *  function of the name. Parse node ip addresses from the filename.
+   * @todo this method introduces mutual dependency between classes {@link H2O} and {@link NetworkInit} ! Move it out!
+   */
   public static void initializeNetworkSockets( ) {
     // Assign initial ports
     H2O.API_PORT = H2O.ARGS.port == 0 ? H2O.ARGS.baseport : H2O.ARGS.port;
 
     // Late instantiation of Jetty object, if needed.
-    if (H2O.getJetty() == null && !H2O.ARGS.disable_web) {
-      H2O.setJetty(new JettyHTTPD());
+    if (H2O.getServletContainer() == null && !H2O.ARGS.disable_web) {
+      H2O.setServletContainer(H2oServletContainerLoader.INSTANCE.createServletContainer());
```

Comment:
should we use `H2OServletContainerLoader` capital o? It violates the camel-case but is more consistent with the rest of the code

Reply:
good point, I will rename it